RadixWallet/Core/FeaturePrelude/AddressView/AddressDetails+View.swift

```diff
@@ -0,0 +1,266 @@
+extension AddressDetails.State {
+	var viewState: AddressDetails.ViewState {
+		.init(
+			address: address,
+			title: title,
+			qrImage: qrImage,
+			showEnlarged: showEnlarged,
+			showShare: showShare
+		)
+	}
+}
+
+// MARK: - AddressDetails.View
+
+public extension AddressDetails {
+	struct ViewState: Equatable {
+		let address: LedgerIdentifiable.Address
+		let colorisedAddress: AttributedString
+		let enlargedAddress: AttributedString
+		let title: Loadable<String?>
+		let qrImage: Loadable<CGImage>
+		let showEnlarged: Bool
+		let showShare: Bool
+
+		init(address: LedgerIdentifiable.Address, title: Loadable<String?>, qrImage: Loadable<CGImage>, showEnlarged: Bool, showShare: Bool) {
+			self.address = address
+			self.colorisedAddress = Self.colorised(address: address)
+			self.enlargedAddress = Self.enlarged(address: address)
+			self.title = title
+			self.qrImage = qrImage
+			self.showEnlarged = showEnlarged
+			self.showShare = showShare
+		}
+	}
+
+	@MainActor
+	struct View: SwiftUI.View {
+		private let store: StoreOf<AddressDetails>
+
+		public init(store: StoreOf<AddressDetails>) {
+			self.store = store
+		}
+
+		public var body: some SwiftUI.View {
+			WithViewStore(store, observe: \.viewState, send: { .view($0) }) { viewStore in
+				content(viewStore)
+					.withNavigationBar {
+						viewStore.send(.closeButtonTapped)
+					}
+					.presentationDetents([.large])
+					.presentationDragIndicator(.visible)
+					.task {
+						viewStore.send(.task)
+					}
+					.sheet(isPresented: viewStore.binding(get: \.showShare, send: .shareDismissed)) {
+						ShareView(items: [viewStore.address.address])
+					}
+			}
+		}
+
+		private func content(_ viewStore: ViewStoreOf<AddressDetails>) -> some SwiftUI.View {
+			VStack(spacing: .zero) {
+				top(title: viewStore.title, showQrCode: viewStore.showQrCode, qrImage: viewStore.qrImage)
+
+				fullAddress(address: viewStore.colorisedAddress)
+					.padding(.vertical, .medium3)
+
+				bottom(showVerifyOnLedger: viewStore.showVerifyOnLedger)
+
+				Spacer()
+			}
+			.multilineTextAlignment(.center)
+			.padding([.horizontal, .bottom], .medium3)
+			.overlay(alignment: .top) {
+				Group {
+					if viewStore.showEnlarged {
+						enlargedView(text: viewStore.enlargedAddress)
+					}
+				}
+				.animation(.easeInOut, value: viewStore.showEnlarged)
+			}
+		}
+
+		private func top(title: Loadable<String?>, showQrCode: Bool, qrImage: Loadable<CGImage>) -> some SwiftUI.View {
```

Comment:
I personally prefer to make these into nested View structs when they take a bunch of parameters. But with observation you shouldn't have to pass anything in, since you can access `store` from the functions.